### PR TITLE
ENH: Adds predefined_split

### PR DESCRIPTION
**Pull request by thomasjpfan**
Fixes https://github.com/dnouri/skorch/issues/282

### CODE REVIEW

File: docs/user/FAQ.rst [hunk 1]
```diff
@@ -110,3 +110,43 @@ dictionary. Below, there is example code on how to achieve this:
...
     net = MyNet(MyModule, ..., criterion__reduce=False)
     net.fit(X, y)
+
+
+I already split my data into training and validation sets, how can I use them?
+------------------------------------------------------------------------------
+
+If you have already defined a training and validation datasets that are
+subclasses of :class:`~torch.utils.data.Dataset`, you can use
+:func:`~skorch.helpers.predefined_split` to train your network:
```
Comment (taketwo) [inline docs/user/FAQ.rst]: Somehow the sentence "you can use `predefined_split()` to train your network" does not sound right to me. We are not using this function to train the network. We are using this function to attach to the network a fixed validation dataset that is independent on the training data passed to the `fit()` method.

Comment (thomasjpfan) [reply docs/user/FAQ.rst]: I see what you mean. I rephrased the sentence to try to make it more clear.

File: docs/user/FAQ.rst [hunk 2]
```diff
@@ -110,3 +110,44 @@ dictionary. Below, there is example code on how to achieve this:
     # for each sample so that it can be weighted
     net = MyNet(MyModule, ..., criterion__reduce=False)
     net.fit(X, y)
+
+
+I already split my data into training and validation sets, how can I use them?
+------------------------------------------------------------------------------
+
+If you have already defined a training and validation datasets that are
```
Comment (ottonemo) [inline docs/user/FAQ.rst]: > [...] defined *a* training [...]

typo

File: docs/user/FAQ.rst [hunk 3]
```diff
@@ -110,3 +110,44 @@ dictionary. Below, there is example code on how to achieve this:
...
+.. code:: python
+
+    from skorch.helpers import predefined_split
+
+    net = NeuralNet(
+        ...,
+        train_split=predefined_split(valid_ds)
+    )
+    # None is passed because train_ds already contains the target
+    net.fit(train_ds, None)
```
Comment (ottonemo) [inline docs/user/FAQ.rst]: for `NeuralNet` you don't need to pass `y=None`, this is only necessary for classifiers and regressors.

File: skorch/tests/test_helper.py [hunk 4]
```diff
@@ -280,3 +281,23 @@ def forward(self, X):
...
+        from skorch.dataset import Dataset
+
+        valid_dataset = Dataset(*data)
+        train_split = predefined_split(valid_dataset)
+
+        tmp_fn = tmpdir.mkdir('predefined_split').join('test.pkl')
+
+        # does not raise
+        with tmp_fn.open('wb') as f:
+            pickle.dump(train_split, f)
```
Comment (ottonemo) [inline skorch/tests/test_helper.py]: Maybe just dump the pickle into a string using `pickle.dumps`? Saves the temporary file hassle.

**Comment by benjamin-work** [comment]
Just generally: Do we want to have `make_split` a public function? What other purpose does it serve? Also: Do we want to have one in `utils.py` and the other in `helper.py` or not both in the same module (which)?

File: skorch/helper.py [hunk 5]
```diff
@@ -113,3 +114,21 @@ def filtered_optimizer(optimizer, filter_fn):
...
+    >>> valid_ds = skorch.Dataset(X, y)
+    >>> net = NeutralNet(..., train_split=predefined_split(valid_ds))
+
+    Parameters
+    ----------
+    dataset: torch Dataset
+       Validiation dataset
+
+    """
+    from skorch.utils import make_split
```
Comment (benjamin-work) [inline skorch/helper.py]: Could the import not be at top level?

**Comment by thomasjpfan** [comment]
`predefined_split` is in `helper.py` since it is meant to accessed by a user.`make_split` is in `utils.py` since it only used by `skorch` internally. 

This PR was updated to convert `make_split` to a private function.

**Pull request merged**